### PR TITLE
Create project browse prompt (close #584)

### LINKED ISSUE #584: I am not prompted what to browse for when creating a new smart contract project
It might be nice to prompt the user to browse to a location to store the new project, in a similar way to the cert/private key prompt when editing a gateway.

I had been side tracked (opening an issue!) part way through the create project task, so it took me a moment to realise what the folder dialog wanted.

### PR DESCRIPTION
**Pull request by Jakeeyturner**
Signed-off-by: Jake Turner <jaketurner25@live.com>

### CODE REVIEW

File: client/src/commands/UserInputUtil.ts [hunk 1]
```diff
@@ -632,6 +632,37 @@ export class UserInputUtil {
 
     }
 
+    public static async saveDirectory(placeHolder: string): Promise<vscode.Uri> {
```
Comment (caroline-church) [inline client/src/commands/UserInputUtil.ts]: This function is very similar to the browseEdit one, can we create more of a common function?

File: client/test/commands/userInputUtil.test.ts [hunk 2]
```diff
@@ -1257,4 +1257,62 @@ describe('userInputUtil', () => {
...
+                canSelectFolders: true,
+                canSelectMany: false,
+                openLabel: 'Save'
+            });
+
+            should.equal(result, undefined);
+        });
+
+        it('should handle any errors', async () => {
+            const logStub: sinon.SinonStub = mySandBox.stub(VSCodeBlockchainOutputAdapter.instance(), 'log');
```
Comment (caroline-church) [inline client/test/commands/userInputUtil.test.ts]: This should be a spy you don't need to stub it

File: client/src/commands/UserInputUtil.ts (hunk 1 above)
Comment (Jakeeyturner) [reply client/src/commands/UserInputUtil.ts]: I've changed the `browseEdit` function and allowed it to be more flexible.

File: client/test/commands/userInputUtil.test.ts (hunk 2 above)
Comment (Jakeeyturner) [reply client/test/commands/userInputUtil.test.ts]: Done!

**Pull request merged**